float64 angle1st
